home/vasco-santos/2021-02-exploration-report-connect-lotus-client-with-deployed-lotus-instance.md

```diff
@@ -0,0 +1,28 @@
+## Connect a Lotus js client to a Lotus running instance
+
+### Background
+
+The [lotus JS client](https://github.com/filecoin-shipyard/js-lotus-client) is a collection of small JS libraries, which enables developers to interact with Lotus via it’s JSON-RPC API. It is decoupled in:
+
+- [Schema](https://github.com/filecoin-shipyard/js-lotus-client-schema): contains schemas for JSON data representation compatible with Lotus JSON-RPC API
+- Provider libraries ([node](https://github.com/filecoin-shipyard/js-lotus-client-provider-nodejs) and [browser](https://github.com/filecoin-shipyard/js-lotus-client-provider-browser)): inspired by “providers” in the Ethereum ecossystem, a provider links to a running node exposing interface that connects to a Lotus JSON-RPC API endpoint using WebSockets or HTTP.
```

Comment:
ahh, so this made me go and look at what the differences are and the nodejs one just wraps the browser one and supplies `ws` and `node-fetch` to polyfill those powers that the browser has natively